src/drivers/memory.js

```diff
@@ -0,0 +1,234 @@
+// If no driver is available, we'll fall back to simple in memory storage.
+// Note that this will have considerable performance and storage
+// side-effects and data will be lost when the user reloads the page
+
+import serializer from '../utils/serializer';
+import Promise from '../utils/promise';
+import executeCallback from '../utils/executeCallback';
+import normalizeKey from '../utils/normalizeKey';
+
+var storageRepository = {};
+
+// Config the backend, using options set in the config.
+function _initStorage(options) {
+    var self = this;
+
+    var dbInfo = {};
+    if (options) {
+        for (var i in options) {
+            dbInfo[i] = options[i];
+        }
+    }
+
+    var database = storageRepository[dbInfo.name] = storageRepository[dbInfo.name] || {};
+    var table = database[dbInfo.storeName] = database[dbInfo.storeName] || {};
+    dbInfo.db = table;
+
+    self._dbInfo = dbInfo;
+    dbInfo.serializer = serializer;
+
+    return Promise.resolve();
+}
+
+// Remove all keys from the datastore, effectively destroying all data in
+// the app's key/value store!
+function clear(callback) {
+    var self = this;
+    var promise = self.ready().then(function() {
+        var db = self._dbInfo.db;
+
+        for (var key in db) {
```

Comment:
For `clear`, couldn't see just set `self._dbInfo.db = {}` rather than manually deleting every key?

Reply:
I think that there was a reason for this at some point, but with the current code that's definitely better :+1:

Reply:
Oh, the reason was that other instances that use the same db will still have access to the old data!
We will have to use a db provider method anyway, so that instances can continue working when their db is dropped.